### PR TITLE
Implemented Infinite Scrolling for Dashboard to Enhance Diagram Browsing Experience

### PR DESCRIPTION
**Pull request by SL-Pirate**
# This PR Introduces Enhanced Interaction and Infinite Scrolling
## Gesture Event Enhancements
- Incorporated gesture detection on arrow bodies, providing a more intuitive interaction experience.
- Introduced secondary tap detections for handlers, significantly improving the desktop browsing experience by allowing right-click interactions.
## Infinite Scrolling for Seamless Navigation
- Implemented infinite scrolling on the dashboard, enabling users to browse diagrams without boundaries, thereby enhancing the overall navigation experience.
## Adjusted Handler Feedback Offset for Improved User Experience
- Modified the handlerFeedbackOffset from Offset(-40, -40) to Offset(0, 0). The previous offset setting was causing unexpected behavior, and the reason for its initial setting was unclear. The adjustment to Offset(0, 0) aligns the handler feedback with the user's interaction point, providing a more intuitive user experience.

### CODE REVIEW

**Comment by SL-Pirate** [comment]
I know this PR has some features that were rejected on #6 
But as far as I can understand it was rejected due to having problems with mobile versions
I don't see such issues in this update.

**Review by SL-Pirate — COMMENTED** [review]
self review: lgtm

File: lib/src/dashboard.dart
Comment (SL-Pirate) [inline lib/src/dashboard.dart:1]: @alnitak Please review this change
`handlerFeedbackOffset = const Offset(-40, -40),`
I'm not sure if this is intentional
but I don't understand why this was set to -40, -40 before.
Nothing seems to be broken after the change

Comment (alnitak) [reply lib/src/dashboard.dart:1]: IIRC that was a way to see the handler when moving it with a finger on a mobile touch screen. When moving you can see the handler shifted from the touch position and not obscured by the finger.

**Comment by alnitak** [comment]
Thank you! Great work!

File: lib/src/dashboard.dart
Comment (SL-Pirate) [reply lib/src/dashboard.dart:1]: I see. Thanks for updating me

**Comment by SL-Pirate** [comment]
> Thank you! Great work!

You are welcome 😄

**Pull request merged**